lang/compiler.go

```diff
@@ -64,12 +155,16 @@ func (c FluxCompiler) CompilerType() flux.CompilerType {
 
 // SpecCompiler implements Compiler by returning a known spec.
 type SpecCompiler struct {
-	Spec *flux.Spec `json:"spec"`
+	Spec  *flux.Spec `json:"spec"`
+	lopts []plan.LogicalOption
+	popts []plan.PhysicalOption
 }
 
 func (c SpecCompiler) Compile(ctx context.Context) (flux.Program, error) {
-	planner := plan.PlannerBuilder{}.Build()
-	ps, err := planner.Plan(c.Spec)
+	planner := plan.PlannerBuilder{}
```

Comment:
nit, the variable being assigned here is not a `Planner`, but a `PlannerBuilder`.  Maybe rename it to `pb`?